lib/galaxy/webapps/galaxy/api/lda_datasets.py

```diff
@@ -194,54 +193,51 @@ def _get_current_roles( self, trans, library_dataset):
         modify_roles = set( trans.app.security_agent.get_roles_for_action( library_dataset, trans.app.security_agent.permitted_actions.LIBRARY_MODIFY ) )
         manage_roles = set( dataset.get_manage_permissions_roles( trans ) )
 
-        access_dataset_role_list = [ access_role.name for access_role in access_roles ]
-        manage_dataset_role_list = [ manage_role.name for manage_role in manage_roles ]
-        modify_item_role_list = [ modify_role.name for modify_role in modify_roles ]
+        access_dataset_role_list = [ ( access_role.name, trans.security.encode_id( access_role.id ) ) for access_role in access_roles ]
+        manage_dataset_role_list = [ ( manage_role.name, trans.security.encode_id( manage_role.id ) ) for manage_role in manage_roles ]
+        modify_item_role_list = [ ( modify_role.name, trans.security.encode_id( modify_role.id ) ) for modify_role in modify_roles ]
 
         return dict( access_dataset_roles=access_dataset_role_list, modify_item_roles=modify_item_role_list, manage_dataset_roles=manage_dataset_role_list )
 
     @expose_api
-    def update_permissions( self, trans, encoded_dataset_id, **kwd ):
+    def update_permissions( self, trans, encoded_dataset_id, payload=None, **kwd ):
         """
-        def update( self, trans, encoded_dataset_id, **kwd ):
-            *POST /api/libraries/datasets/{encoded_dataset_id}/permissions
+        *POST /api/libraries/datasets/{encoded_dataset_id}/permissions
+            Set permissions of the given dataset to the given role ids.
 
         :param  encoded_dataset_id:      the encoded id of the dataset to update permissions of
         :type   encoded_dataset_id:      an encoded id string
-
-        :param  action:     (required) describes what action should be performed
-                            available actions: make_private, remove_restrictions, set_permissions
-        :type   action:     string
-
-        :param  access_ids[]:      list of Role.name defining roles that should have access permission on the dataset
-        :type   access_ids[]:      string or list
-        :param  manage_ids[]:      list of Role.name defining roles that should have manage permission on the dataset
-        :type   manage_ids[]:      string or list
-        :param  modify_ids[]:      list of Role.name defining roles that should have modify permission on the library dataset item
-        :type   modify_ids[]:      string or list
-
-        :rtype:     dictionary
+        :param   payload: dictionary structure containing:
+            :param  action:     (required) describes what action should be performed
+                                available actions: make_private, remove_restrictions, set_permissions
+            :type   action:     string
+            :param  access_ids[]:      list of Role.id defining roles that should have access permission on the dataset
```

Comment:
@nsoranzo this now accepts encoded Role.id instead of Role.name thus changing the endpoint signature considerably :/